sotodlib/data/load.py

```diff
@@ -0,0 +1,317 @@
+        set.
+      obs_name (str): The identifier of the observation.
+      dets (list of str): The detector names of interest.  If None,
+        loads all dets present in this observation.
+      prefix (str): The root address of the data files.  If not
+        specified, the prefix is taken from the ObsFileDB.
+
+    Returns:
+      (signal_streams, other_streams).
+
+    """
+    if prefix is None:
+        prefix = db.prefix
+        if prefix is None:
+            prefix = './'
+
+    # Regardless of what dets have been asked for (maybe none), get
+    # the list of detsets implicated in this observation.
+    c = db.conn.execute('select distinct DS.name, DS.det from detsets DS '
+                        'join files on DS.name=files.detset '
+                        'where obs_id=?', (obs_name,))
+    pairs = [tuple(r) for r in c.fetchall()]
+
+    # Now filter to only the dets requested.
+    if dets is None:
+        pairs_req = pairs
+    else:
+        pairs_req = [p for p in pairs if p[1] in dets]
+        # Use sets for this...
+        dets_req = [p[1] for p in pairs_req]
+        unmatched = [d for d in dets if not d in dets_req]
+        if len(unmatched):
+            raise RuntimeError("User requested invalid dets (e.g. %s) for obs=%s" %
+                               unmatched[0], obs_name)
+
+    # Group by detset.
+    dets_by_detset = OrderedDict([(p[0],[]) for p in pairs_req])
+    for p in pairs_req:
+        dets_by_detset[p[0]].append(p[1])
+
+    # Loop through relevant files, in sample order, and accumulate
+    # lists of stream segments.
+    stream_groups = []
+    for detset, dets in dets_by_detset.items():
+        c = db.conn.execute('select name from files '
+                            'where obs_id=? and detset=? '
+                            'order by sample_start', (obs_name, detset))
+        streams = None
+        request = FieldGroup('root', [
+            FieldGroup('signal', dets, timestamp_field='timestamps',
+                       compression=True),
+            FieldGroup('boresight', ['az', 'el', 'roll']),
+            'site_position',
+            'site_velocity',
+            'boresight_azel',
+            'boresight_radec',
+            #'flags_common',
+        ])
+        for row in c:
+            f = row[0]
+            streams = unpack_frames(prefix+f, request, streams)
+        stream_groups.append((request, streams))
+
+    # Merge the groups.
+    streams = OrderedDict()
+    for request, s in stream_groups:
+        request.merge_result(streams, s)
+        
+    FieldGroup.hstack_result(streams)
+
+    if not toast:
+        return streams
+
+    #
+    # Re-process streams into standard TOD.cache info.
+    #
+    from toast.tod import TOD
+
+    detquats = []
+    nsamp = len(streams['boresight']['az'])
+    dets = list(streams['signal'].keys())
+    detranks = 1
+    sampsizes = [(0, nsamp)]
+    meta = {}
+    tod = TOD(mpicomm, dets, nsamp, detranks=detranks, meta=meta)
```

Comment:
These might be more just questions on TOAST than comments on this. But the comments in `toast.TOD` say to never directly instantiate it like this ( https://github.com/tskisner/toast/blob/refactor_2.3/src/toast/tod/tod.py#L34   ) because of all the `NotImplementedError` exceptions that are there. Are those going to be a problem here?

There's also another TOD object made there? https://github.com/simonsobs/sotodlib/blob/toast_refactor/sotodlib/data/toast_load.py#L38 

It seems like we should have one base TOD object for SO, at minimum one base TOD object for loading from SO "data." Do we have one of these yet?